pkg/sdk/databases_test.go

```diff
@@ -1,258 +1,577 @@
+	t.Run("with set tag", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.SetTag = []TagAssociation{
+			{
+				Name:  NewSchemaObjectIdentifier("db", "schema", "tag1"),
+				Value: "v1",
+			},
+			{
+				Name:  NewSchemaObjectIdentifier("db", "schema", "tag2"),
+				Value: "v2",
 			},
 		}
-		assertOptsValidAndSQLEquals(t, opts, `ALTER DATABASE "db1" SET DATA_RETENTION_TIME_IN_DAYS = 1, COMMENT = 'comment'`)
+		assertOptsValidAndSQLEquals(t, opts, `ALTER DATABASE %s SET TAG "db"."schema"."tag1" = 'v1', "db"."schema"."tag2" = 'v2'`, opts.name.FullyQualifiedName())
 	})
 
-	t.Run("unset comment", func(t *testing.T) {
-		opts := &AlterDatabaseOptions{
-			name: NewAccountObjectIdentifier("db1"),
-			Unset: &DatabaseUnset{
-				Comment: Bool(true),
-			},
+	t.Run("with unset tag", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.UnsetTag = []ObjectIdentifier{
+			NewSchemaObjectIdentifier("db", "schema", "tag1"),
 		}
-		assertOptsValidAndSQLEquals(t, opts, `ALTER DATABASE "db1" UNSET COMMENT`)
+		assertOptsValidAndSQLEquals(t, opts, `ALTER DATABASE %s UNSET TAG "db"."schema"."tag1"`, opts.name.FullyQualifiedName())
 	})
 }
 
 func TestDatabasesAlterReplication(t *testing.T) {
+	defaultOpts := func() *AlterDatabaseReplicationOptions {
+		return &AlterDatabaseReplicationOptions{
+			name: randomAccountObjectIdentifier(),
+		}
+	}
+
+	t.Run("validation: invalid name", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.name = NewAccountObjectIdentifier("")
+		assertOptsInvalidJoinedErrors(t, opts, ErrInvalidObjectIdentifier)
+	})
+
+	t.Run("validation: exactly one action", func(t *testing.T) {
+		opts := defaultOpts()
+		assertOptsInvalidJoinedErrors(t, opts, errExactlyOneOf("AlterDatabaseReplicationOptions", "EnableReplication", "DisableReplication", "Refresh"))
+	})
+
+	t.Run("validation: exactly one action", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.EnableReplication = &EnableReplication{}
+		opts.DisableReplication = &DisableReplication{}
+		assertOptsInvalidJoinedErrors(t, opts, errExactlyOneOf("AlterDatabaseReplicationOptions", "EnableReplication", "DisableReplication", "Refresh"))
+	})
+
 	t.Run("enable replication", func(t *testing.T) {
-		opts := &AlterDatabaseReplicationOptions{
-			name: NewAccountObjectIdentifier("db1"),
-			EnableReplication: &EnableReplication{
-				ToAccounts: []AccountIdentifier{
-					NewAccountIdentifierFromAccountLocator("account1"),
-				},
-				IgnoreEditionCheck: Bool(true),
+		opts := defaultOpts()
+		opts.EnableReplication = &EnableReplication{
+			ToAccounts: []AccountIdentifier{
+				NewAccountIdentifierFromAccountLocator("account1"),
 			},
+			IgnoreEditionCheck: Bool(true),
 		}
-		assertOptsValidAndSQLEquals(t, opts, `ALTER DATABASE "db1" ENABLE REPLICATION TO ACCOUNTS "account1" IGNORE EDITION CHECK`)
+		assertOptsValidAndSQLEquals(t, opts, `ALTER DATABASE %s ENABLE REPLICATION TO ACCOUNTS "account1" IGNORE EDITION CHECK`, opts.name.FullyQualifiedName())
```

Comment:
Just curioius, why replication and failover operations are separate from alter?

Reply:
I think because it was
- Already that way and I don't think it would be beneficial to join those alters right now especially, because I'm pretty certain replication and failover cannot be invoked with other alter calls like SET COMMENT which would make it only a visual change.
- Also separated in the documentation